File: src/main/java/me/nobaboy/nobaaddons/keybinds/NobaKeyBind.java
package me.nobaboy.nobaaddons.keybinds;

import me.nobaboy.nobaaddons.NobaAddons;
import me.nobaboy.nobaaddons.util.CooldownManager;
import net.minecraft.client.settings.KeyBinding;
import org.lwjgl.input.Keyboard;

public class NobaKeyBind extends KeyBinding {
	private final CooldownManager cooldown = new CooldownManager();
	private final Runnable onPress;

	public NobaKeyBind(String name, int key, Runnable onPress) {
		super(name, key, NobaAddons.MOD_NAME);
		this.onPress = onPress;
	}

	public NobaKeyBind(String name, Runnable onPress) {
		this(name, Keyboard.KEY_NONE, onPress);
	}

	public void maybePress() {
		if(this.isPressed() && !cooldown.isOnCooldown()) {
			cooldown.startCooldown(500L);
			onPress.run();
		}
	}
}

File: src/main/kotlin/me/nobaboy/nobaaddons/keybinds/NobaKeyBind.kt
package me.nobaboy.nobaaddons.keybinds

import me.nobaboy.nobaaddons.NobaAddons
import me.nobaboy.nobaaddons.util.CooldownManager
import net.minecraft.client.settings.KeyBinding
import org.lwjgl.input.Keyboard

open class NobaKeyBind(
    name: String, key: Int = Keyboard.KEY_NONE, private val onPress: () -> Unit
) : KeyBinding(name, key, NobaAddons.MOD_NAME) {
    private val cooldown = CooldownManager()

    constructor(name: String, onPress: () -> Unit) : this(name, Keyboard.KEY_NONE, onPress)

    fun maybePress() {
        if (this.isPressed && !cooldown.isOnCooldown()) {
            cooldown.startCooldown(500L)
            onPress()
        }
    }
}